frontend/src/pages/connectionTypes/ConnectionTypesTableRow.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { Td, Tr } from '@patternfly/react-table';
+import {
+  Flex,
+  Icon,
+  Label,
+  Switch,
+  Text,
+  TextContent,
+  Timestamp,
+  TimestampTooltipVariant,
+  Tooltip,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { ConnectionTypeConfigMapObj } from '~/concepts/connectionTypes/types';
+import { relativeTime } from '~/utilities/time';
+import { updateConnectionTypeEnabled } from '~/services/connectionTypesService';
+
+type ConnectionTypesTableRowProps = {
+  obj: ConnectionTypeConfigMapObj;
+  onUpdate: () => void;
+};
+
+const ConnectionTypesTableRow: React.FC<ConnectionTypesTableRowProps> = ({ obj, onUpdate }) => {
+  const [statusMessage, setStatusMessage] = React.useState<string | undefined>();
+  const [errorMessage, setErrorMessage] = React.useState<string | undefined>();
+  const pendingEnabledState = React.useRef<'true' | 'false' | undefined>();
+  const createdDate = obj.metadata.creationTimestamp
+    ? new Date(obj.metadata.creationTimestamp)
+    : undefined;
+
+  const onUpdateEnabled = async (enabled: boolean) => {
+    setStatusMessage(enabled ? 'Enabling...' : 'Disabling...');
+    setErrorMessage(undefined);
+    pendingEnabledState.current = enabled ? 'true' : 'false';
+
+    const response = await updateConnectionTypeEnabled(obj, enabled);
+    if (response.success) {
+      onUpdate();
+      return;
+    }
```

Comment:
Maybe have `onUpdate` return a promise to remove the reference and useEffect for managing this toggle message state?